### PR TITLE
Vanilla Kube Capability: Non-root Security Context

### PR DESCRIPTION
**Pull request by gsrajpal**
Set Non-root Security Context for Vanilla Kubernetes cluster

- This PR leverages Management Cluster Capability to determine SCC feature and adds `securityContext`

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/clusterpolicy/deployment.go [hunk 1]
```diff
@@ -71,7 +71,6 @@ func buildOCMContainerMain(image string) func(*corev1.Container) {
 			path.Join(volumeMounts.Path(c.Name, cpcVolumeConfig().Name), configKey),
 			"--kubeconfig",
 			path.Join(volumeMounts.Path(c.Name, cpcVolumeKubeconfig().Name), kas.KubeconfigKey),
-			"--namespace=openshift-kube-controller-manager",
```
Comment (alvaroaleman) [inline control-plane-operator/controllers/hostedcontrolplane/clusterpolicy/deployment.go:74]: Why was this removed?

File: control-plane-operator/controllers/hostedcontrolplane/manifests/pki.go [hunk 2]
```diff
@@ -131,15 +131,6 @@ func OpenShiftOAuthAPIServerCertSecret(ns string) *corev1.Secret {
 	}
 }
 
-func OpenshiftAuthenticatorCertSecret(ns string) *corev1.Secret {
```
Comment (alvaroaleman) [inline control-plane-operator/controllers/hostedcontrolplane/manifests/pki.go:134]: This seems unrelated

File: control-plane-operator/controllers/hostedcontrolplane/manifests/pki.go [hunk 3]
```diff
@@ -230,19 +221,19 @@ func OLMPackageServerCertSecret(ns string) *corev1.Secret {
 	}
 }
 
-func OLMProfileCollectorCertSecret(ns string) *corev1.Secret {
+func OLMOperatorServingCertSecret(ns string) *corev1.Secret {
```
Comment (alvaroaleman) [inline control-plane-operator/controllers/hostedcontrolplane/manifests/pki.go:224]: This also seems unrelated

File: control-plane-operator/controllers/hostedcontrolplane/manifests/pki.go [hunk 4]
```diff
@@ -230,19 +221,19 @@ func OLMPackageServerCertSecret(ns string) *corev1.Secret {
...
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "olm-profile-collector",
+			Name:      "olm-operator-serving-cert",
 			Namespace: ns,
 		},
 	}
 }
 
-func OLMOperatorServingCertSecret(ns string) *corev1.Secret {
+func OpenshiftAuthenticatorCertSecret(ns string) *corev1.Secret {
```
Comment (alvaroaleman) [inline control-plane-operator/controllers/hostedcontrolplane/manifests/pki.go:233]: Unrelated

File: control-plane-operator/controllers/hostedcontrolplane/configoperator/params.go [hunk 5]
```diff
@@ -43,6 +44,15 @@ func NewHostedClusterConfigOperatorParams(ctx context.Context, hcp *hyperv1.Host
 			},
 		},
 	}
+
+	if explicitNonRootSecurityContext {
```
Comment (alvaroaleman) [inline control-plane-operator/controllers/hostedcontrolplane/configoperator/params.go:48]: Do we ever have a container that should not have the RunAsUser? If not, we should probably instead just iterate over all of them, this also has the advantage that it will do the right thing in the future if we add additional containers

**Review by alvaroaleman — COMMENTED** [review]
Generally looks good, but if possible I'd prefer an approach where we simply set this for all containers. Also, the PR needs cleaning up, there is a bunch of unrelated changes and it pulls in 97 unrelated commits

**Comment by gsrajpal** [comment]
@alvaroaleman I will just close this and create a new PR with the needed changes. 

Agreed! too many unnecessary commits.
